# GitLab NGINX Patch: Allow to configure priority class names with global
# chart configuration.
diff --git a/charts/nginx-ingress/templates/admission-webhooks/job-patch/job-createSecret.yaml b/charts/nginx-ingress/templates/admission-webhooks/job-patch/job-createSecret.yaml
index af3ea12a3..463ab80d7 100644
--- a/charts/nginx-ingress/templates/admission-webhooks/job-patch/job-createSecret.yaml
+++ b/charts/nginx-ingress/templates/admission-webhooks/job-patch/job-createSecret.yaml
@@ -34,8 +34,8 @@ spec:
         {{- toYaml . | nindent 8 }}
         {{- end }}
     spec:
-    {{- if .Values.controller.admissionWebhooks.patch.priorityClassName }}
-      priorityClassName: {{ .Values.controller.admissionWebhooks.patch.priorityClassName }}
+    {{- if or $.Values.global.priorityClassName .Values.controller.admissionWebhooks.patch.priorityClassName }}
+      priorityClassName: {{ default $.Values.global.priorityClassName .Values.controller.admissionWebhooks.patch.priorityClassName }}
     {{- end }}
     {{- if .Values.imagePullSecrets }}
       imagePullSecrets: {{ toYaml .Values.imagePullSecrets | nindent 8 }}
diff --git a/charts/nginx-ingress/templates/admission-webhooks/job-patch/job-patchWebhook.yaml b/charts/nginx-ingress/templates/admission-webhooks/job-patch/job-patchWebhook.yaml
index 87dd2c251..862c83c1e 100644
--- a/charts/nginx-ingress/templates/admission-webhooks/job-patch/job-patchWebhook.yaml
+++ b/charts/nginx-ingress/templates/admission-webhooks/job-patch/job-patchWebhook.yaml
@@ -34,8 +34,8 @@ spec:
         {{- toYaml . | nindent 8 }}
         {{- end }}
     spec:
-    {{- if .Values.controller.admissionWebhooks.patch.priorityClassName }}
-      priorityClassName: {{ .Values.controller.admissionWebhooks.patch.priorityClassName }}
+    {{- if or .Values.global.priorityClassName .Values.controller.admissionWebhooks.patch.priorityClassName }}
+      priorityClassName: {{ default .Values.global.priorityClassName .Values.controller.admissionWebhooks.patch.priorityClassName }}
     {{- end }}
     {{- if .Values.imagePullSecrets }}
       imagePullSecrets: {{ toYaml .Values.imagePullSecrets | nindent 8 }}
diff --git a/charts/nginx-ingress/templates/controller-daemonset.yaml b/charts/nginx-ingress/templates/controller-daemonset.yaml
index d9f69a4e0..d55d5e885 100644
--- a/charts/nginx-ingress/templates/controller-daemonset.yaml
+++ b/charts/nginx-ingress/templates/controller-daemonset.yaml
@@ -56,8 +56,8 @@ spec:
     {{- if .Values.imagePullSecrets }}
       imagePullSecrets: {{ toYaml .Values.imagePullSecrets | nindent 8 }}
     {{- end }}
-    {{- if .Values.controller.priorityClassName }}
-      priorityClassName: {{ .Values.controller.priorityClassName | quote }}
+    {{- if or .Values.global.priorityClassName .Values.controller.priorityClassName }}
+      priorityClassName: {{ default .Values.global.priorityClassName .Values.controller.priorityClassName }}
     {{- end }}
     {{- if or .Values.controller.podSecurityContext .Values.controller.sysctls }}
       securityContext:
diff --git a/charts/nginx-ingress/templates/controller-deployment.yaml b/charts/nginx-ingress/templates/controller-deployment.yaml
index 3c8b79296..4fc313448 100644
--- a/charts/nginx-ingress/templates/controller-deployment.yaml
+++ b/charts/nginx-ingress/templates/controller-deployment.yaml
@@ -67,8 +67,8 @@ spec:
     {{- if .Values.imagePullSecrets }}
       imagePullSecrets: {{ toYaml .Values.imagePullSecrets | nindent 8 }}
     {{- end }}
-    {{- if .Values.controller.priorityClassName }}
-      priorityClassName: {{ .Values.controller.priorityClassName | quote }}
+    {{- if or $.Values.global.priorityClassName .Values.controller.priorityClassName }}
+      priorityClassName: {{ default $.Values.global.priorityClassName .Values.controller.priorityClassName }}
     {{- end }}
     {{- if or .Values.controller.podSecurityContext .Values.controller.sysctls }}
       securityContext:
diff --git a/charts/nginx-ingress/templates/default-backend-deployment.yaml b/charts/nginx-ingress/templates/default-backend-deployment.yaml
index d242f053d..ee562008e 100644
--- a/charts/nginx-ingress/templates/default-backend-deployment.yaml
+++ b/charts/nginx-ingress/templates/default-backend-deployment.yaml
@@ -48,8 +48,8 @@ spec:
     {{- if .Values.imagePullSecrets }}
       imagePullSecrets: {{ toYaml .Values.imagePullSecrets | nindent 8 }}
     {{- end }}
-    {{- if .Values.defaultBackend.priorityClassName }}
-      priorityClassName: {{ .Values.defaultBackend.priorityClassName }}
+    {{- if or .Values.global.priorityClassName .Values.defaultBackend.priorityClassName }}
+      priorityClassName: {{ default .Values.global.priorityClassName .Values.defaultBackend.priorityClassName }}
     {{- end }}
     {{- if .Values.defaultBackend.podSecurityContext }}
       securityContext: {{ toYaml .Values.defaultBackend.podSecurityContext | nindent 8 }}
